chef-config/lib/chef-config/config.rb

```diff
@@ -796,6 +796,15 @@ def self.windows_home_path
     config_context :chefdk do
     end
 
+    # Configuration options for Data Collector reporting. These settings allow
+    # the user to configure where to send their Data Collector data, what token
+    # to send, and whether Data Collector should report its findings in client
+    # mode vs. solo mode.
+    default :data_collector_server_url,       nil
```

Comment:
It might be better for our namespaces to do this:

```
config_context :data_collector do
  default :server_url, nil
  default :token, nil
  default :mode, :both
  default :raise_on_failure, false
end
```

They can then be set in the config file with

```
data_collector.server_url = "https://..."
data_collector.mode = :both
```

etc.

Also, please comment new config values with acceptable values and what they affect.